cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceConnector.java

```diff
@@ -102,6 +103,13 @@
         return awsLaunchService.launch(ac, stack, resourceNotifier, adjustmentType, threshold);
     }
 
+    @Override
+    public List<CloudResourceStatus> launchLoadBalancer(AuthenticatedContext authenticatedContext, CloudStack stack, PersistenceNotifier persistenceNotifier)
```

Comment:
**question:** Why are we adding this `launchLoadBalancer` method? I suppose it makes sense conceptually to include a separate API handle for it, but we generally don't have that for other "parts" of the cloud stack.

The `launch` method just handles the whole `CloudStack` object, of which the load balancers are a part.

Reply:
I can see the utility of something like this when we implement an _upgrade_ path for LBs though. @hreeve-cloudera probably has better ideas here than I do.

Reply:
At least for the Yarn case this was kind of necessary to get all of the logic in the right place. Essentially, we wanted to be able to launch the loadbalancer after the other containers were launched so we could get their IP addresses, but still have the logic in a place where we had access to the stack and could persists the resource we would create for the LB, and this was the best place for it.
Haley mentioned that this would be better in order to isolate the logic completely to help with future changes and the like.

Reply:
We do have an upgrade path for load balancers. Bryce, you and I have talked about it briefly. But basically in CB-10612 I added a new API that, if called, will add load balancers to an existing environment. As part of that I separated out the AWS load balancer logic into a new `AwsLoadBalancerLaunchService`. So for AWS rolling with this change will be simple, because we can simply move where `AwsLoadBalancerLaunchService` is being currently called to be called here instead.

Ultimately we need to be able to call the LB creation logic separately of the stack creation anyway, to support that in-place upgrade API. So I didn't see any harm in separating things out like that, and told Santiago to go for it, and we would figure out how to implement it across the other cloud platforms as necessary.

Reply:
Actually, Santiago, I don't think you need a new method here. This is functionally just a duplicate of the `updateLoadBalancers` method I already added to the `ResourceConnector` interface. So let's consolidate that. (We can rename the existing method to `launchLoadBalancer` if you like, to match the other `launch` method.)

Reply:
When in the flow is that called? It might require a little bit of work to get it working in that order, but not sure.